src/core/Context.py

```diff
@@ -64,32 +69,48 @@ async def prompt(self, message, title=None, delete_after=True):
             embed.title = title
 
         view = utils.Prompt(self.author.id)
-        msg = await self.send(embed=embed, view=view)
+        msg: Optional[discord.Message] = await self.send(embed=embed, view=view)
         await view.wait()
 
         try:
-            if delete_after:
+            if delete_after and msg is not None:
                 await msg.delete(delay=0)
         finally:
             return view.value
 
-    async def error(self, message, delete_after=None, **kwargs) -> discord.Message:
+    async def error(
+        self, message: str, delete_after: bool = None, **kwargs: Any
+    ) -> Optional[discord.Message]:
         with suppress(discord.HTTPException):
 
-            return await self.reply(
+            msg: Optional[discord.Message] = await self.reply(
                 embed=discord.Embed(description=message, color=discord.Color.red()),
                 delete_after=delete_after,
                 **kwargs,
             )
+            try:
+                await self.bot.wait_for(
+                    "message_delete", check=lambda m: m.id == self.message.id, timeout=30
+                )
+            except asyncio.TimeoutError:
+                pass
+            else:
+                if msg is not None:
+                    await msg.delete(delay=0)
+            finally:
+                return msg
```

Comment:
Idea behind is, error messages looks awful no matter how decorated it is. Within a given time interval if the original message got deleted (`Context.message`) then bot will delete its message (`msg`).

---

**Use case**: More neat and tidy `#lobby` `#general`.

---

This won't break the bot as you never used the `Message` object returned from `ctx.error` anywhere in the code